modules/platform/nuxeo-platform-content-template-manager/src/main/java/org/nuxeo/ecm/platform/content/template/factories/SimpleTemplateBasedRootFactory.java

```diff
@@ -37,8 +37,10 @@
     public void createContentStructure(DocumentModel eventDoc) {
         initSession(eventDoc);
 
+        boolean initRootAcl = true;
         for (TemplateItemDescriptor item : template) {
             if (!shouldCreateDocument(eventDoc.getId(), item.getId(), item.getTypeName())) {
+                initRootAcl = false;
```

Comment:
Shouldn't it be the opposite? Start with false and if we found one to create set it to true?

Reply:
thx for the point, 
My idea here is to keep the same flow that was before my initial fix see [here](https://github.com/nuxeo/nuxeo/commit/8f84864a3bcdf4190b72409d0ff194929b471f85)

before the fix above, the acl was computed/inited only once (call to `shouldCreateContent` returns true if at least once child exist) this is the reason why we didn't see the problem before.

Reply:
Ok, but the flow has been changed with the fix. Now it makes more sense to me to update root ACLs if a child was created whether to not update them if we didn't create a child, or I'm missing something.

Do you know what happen in such cases? Ie: a child is created and another one is skipped which results in not setting root acls.

Reply:
I see your point of view.I was thinking the same thing. 

>Do you know what happen in such cases? Ie: a child is created and another one is skipped which results in not setting root acls.

>...Ie: a child is created and another one is skipped

with introduced fix each child is created, but the acl of root is re-computed each time. even they don't change, in the case of MongoDB it will re-compute all acl of the children but without making  a merge which it means that the acl are removed and re-created from the root onces and if I am not wrong we will even loose the acl defined on each document.

Reply:
Ok, is it the case of acls or just read calls?

Are you saying the last set acl is not useful?